From 2ea5b5aec420781de0b7cdefec5a4d0d43982682 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Tue, 16 Jun 2020 11:31:01 -0500
Subject: [PATCH 4/6] configs: beacon-rzg2m.h: Simplify Boot scripts

Other boards from Logic PD/Beacon EmbeddeWorks use boot scripts
to allow the user to select the mmc device, then 'boot' without
having to change bootargs, loading partitions, etc.

This patch migrates these scripts to the RZ/G2M kit.  The default
is currently setup to boot from mmc 0 which is the micro SD card
on the baseboard.  It expects two partitions with the first
containing the Image and the dtb file.

Default mmcdev:0 (uSD)
The default fdt_file is:  r8a774a1-beacon-rzg2m-kit.dtb

Assuming the eMMC is programmed with the same partition structure,
Booting from eMMC is done wtih the following:

setenv mmcdev 1
saveenv (optional)
boot

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 include/configs/beacon-rzg2m.h | 73 ++++++++++++++++++++++++++++++++++
 1 file changed, 73 insertions(+)

diff --git a/include/configs/beacon-rzg2m.h b/include/configs/beacon-rzg2m.h
index f5b229ec2b..e788a4ee82 100644
--- a/include/configs/beacon-rzg2m.h
+++ b/include/configs/beacon-rzg2m.h
@@ -20,4 +20,77 @@
 #define CONFIG_SYS_MMC_ENV_DEV		1
 #define CONFIG_SYS_MMC_ENV_PART		2
 
+#undef CONFIG_EXTRA_ENV_SETTINGS
+
+#define CONFIG_EXTRA_ENV_SETTINGS		\
+	"usb_pgood_delay=2000\0"	\
+	"fdt_high=0xffffffffffffffff\0"	\
+	"initrd_high=0xffffffffffffffff\0" \
+	"script=boot.scr\0" \
+	"image=Image\0" \
+	"console=ttySC0,115200\0" \
+	"fdt_addr=0x48080000\0"			\
+	"fdt_high=0xffffffffffffffff\0"		\
+	"boot_fdt=try\0" \
+	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
+	"initrd_addr=0x43800000\0"		\
+	"initrd_high=0xffffffffffffffff\0" \
+	"mmcdev=0\0" \
+	"mmcpart=1\0" \
+	"mmcrootpart=2\0" \
+	"finduuid=part uuid mmc ${mmcdev}:${mmcrootpart} uuid\0" \
+	"mmcautodetect=yes\0" \
+	"mmcargs=setenv bootargs console=${console} " \
+	" root=PARTUUID=${uuid} rootwait rw ${optargs}\0" \
+	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
+	"bootscript=echo Running bootscript from mmc ...; " \
+		"source\0" \
+	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
+	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"mmcboot=echo Booting from mmc ...; " \
+		"run finduuid; run mmcargs; " \
+		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
+			"if run loadfdt; then " \
+				"booti ${loadaddr} - ${fdt_addr}; " \
+			"else " \
+				"echo WARN: Cannot load the DT; " \
+			"fi; " \
+		"else " \
+			"echo wait for boot; " \
+		"fi;\0" \
+	"netargs=setenv bootargs ${jh_clk} console=${console} " \
+		"root=/dev/nfs " \
+		"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0" \
+	"netboot=echo Booting from net ...; " \
+		"run netargs;  " \
+		"if test ${ip_dyn} = yes; then " \
+			"setenv get_cmd dhcp; " \
+		"else " \
+			"setenv get_cmd tftp; " \
+		"fi; " \
+		"${get_cmd} ${loadaddr} ${image}; " \
+		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
+			"if ${get_cmd} ${fdt_addr} ${fdt_file}; then " \
+				"booti ${loadaddr} - ${fdt_addr}; " \
+			"else " \
+				"echo WARN: Cannot load the DT; " \
+			"fi; " \
+		"else " \
+			"booti; " \
+		"fi;\0"
+
+#undef CONFIG_BOOTCOMMAND
+
+#define CONFIG_BOOTCOMMAND \
+	"mmc dev ${mmcdev}; if mmc rescan; then " \
+	   "if run loadbootscript; then " \
+		   "run bootscript; " \
+	   "else " \
+		   "if run loadimage; then " \
+			   "run mmcboot; " \
+		   "else run netboot; " \
+		   "fi; " \
+	   "fi; " \
+	"else booti ${loadaddr} - ${fdt_addr}; fi"
+
 #endif /* __BEACON_RZG2M_H */
-- 
2.17.1

